src/renderer/actions.ts

```diff
@@ -490,6 +489,7 @@ export function loadPreferences(): ThunkAction {
             dispatch(loadInitialWorkspace(
                 getState().session.reopenLastWorkspace,
                 getState().session.lastWorkspacePath));
+            dispatch(loadDataStores());
```

Comment:
Loading data stores after loading workspace doesn't seem right. That way, opening datasets from last opened workspace will not be possible.

Reply:
I suggest finding the root cause (you might have come close) then solve the root problem.

Reply:
The problem occurs here: [dispatch(loadDataStores());](https://github.com/CCI-Tools/cate-app/blob/85e155aceda00de09de077f4544e53d7c4f38382/src/renderer/actions.ts#L971)

If the previous session has not been loaded yet, the session value of selectedDataStoreId is null and set to the id of the first list entry - which is why you will not encounter this issue for the first data store (cci-store). The solution is to load in the data stores after the session has been loaded.